gateway-sp-comms/src/error.rs

```diff
@@ -85,16 +85,18 @@ pub enum UpdateError {
     ImageEmpty,
     #[error("update image is too large")]
     ImageTooLarge,
-    #[error("hubris archive error: {0}")]
+    #[error("hubris archive error")]
     HubtoolsError(#[from] hubtools::Error),
-    #[error("error reading caboose in hubris archive: {0}")]
+    #[error("error reading caboose in hubris archive")]
     CabooseError(#[from] hubtools::CabooseError),
     #[error("board mismatch: SP is {sp} but archive is for {archive}")]
     BoardMismatch { sp: String, archive: String },
+    #[error("RoT slot mismatch: target slot {slot} cannot be written with {image_name:?} image")]
+    RotSlotMismatch { slot: u16, image_name: String },
```

Comment:
We're allowed to freely reorder stuff in this enum, right?

Reply:
Yes, this one is in "infinite memory machine" land and is not serialized via hubpack.